### ISSUE TITLE
VRam deadlocking solution

### ISSUE THREAD

**Issue by DemonSkye** [d8ahazard/sd_dreambooth_extension#746]
**WHAT**
At the moment, there is no smart memory management and a system needs to be nearly idle for the training of a model to perform.  I have a 3090 which has 24gb of Vram, and despite the command line suggesting that I'm using 6.0/22.3gb of it, I left my computer unattended to train a model at about 4:30pm today.   I came back to find that the model was stuck on step 637, and that I had a slack notification come in, which for some reason demanded SOME amount of vram, which it was able to claim.   The step that was in progress demanded more Vram, neither process could complete and my system hung in a state like that for 5.5 hours while I was out running errands.   There should be a backoff method or a temporary cache to ram to attempt to free up deadlocking in scenarios like this.   As you're all aware, slack isn't a GPU intensive application.   It was likely trying to raise a systray notification or something, or paint something into a minimized app, which would've been <1 second of GPU time.  

While I understand that this process is demanding on GPU, I will also note that I'm running this with --medvram and set PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.6,max_split_size_mb:128, so that idea that I'm running out of VRam on a 3090 is crazy to begin with.

I've had similar hangs due to my desktop wallpaper manager (wallpaper engine) which was essentially idle at the time.

**Comment by AyoKeito** [comment]
Can confirm this happens sometimes on 4090 with the following settings:
> Attention: xformers
> Gradient Checkpointing: True
> Accumulation Steps: 7
> Precision: fp16
> Cache Latents: True
> Training Batch Size: 7
> Class Generation Batch Size: 1
> Text Encoder Ratio: 0.75
> 8Bit Adam: False
> EMA: True
> LORA: False

Training just stops with no errors shortly after i lock the screen. Resumes immediately after i unlock it. No errors.
Partial workaround for me is to close **EVERYTHING** other than dreambooth, including browser (it's a shame, because them i'm unable to see the stats or cancel from webui) **and** reduce `Save Preview(s) Frequency`

Maybe related to previews generation?

**Comment by d8ahazard** [comment]
Give it a pull and test now.
